app/workflows/same_appeal_substitution_tasks_factory.rb

```diff
@@ -20,11 +22,63 @@ def create_substitute_tasks!
   def create_tasks_for_distributed_appeal
     if @appeal.docket_type == Constants.AMA_DOCKETS.hearing && selected_tasks_include_hearing_tasks?
       send_hearing_appeal_back_to_distribution
+    elsif evidence_submission_task_selected?
+      resume_evidence_submission
     elsif no_tasks_selected?
       reopen_decision_tasks
     end
   end
 
+  def evidence_submission_task_selected?
+    selected_tasks = Task.where(id: @task_ids[:selected]).order(:id)
+    !selected_tasks.of_type(:EvidenceSubmissionWindowTask).empty?
+  end
+
+  ATTRIBUTES_EXCLUDED_FROM_TASK_COPY = %w[id created_at updated_at
+                                          status closed_at placed_on_hold_at].freeze
+
+  def copy_esw_task_with_ancestors(task)
+    parent = task.parent
+    return unless parent
+
+    new_task_attributes = task
+      .attributes
+      .except(*ATTRIBUTES_EXCLUDED_FROM_TASK_COPY, ["parent_id"])
+
+    existing_new_parent = @appeal.reload.tasks.open.find { |option| option.type == parent.type }
```

Comment:
```suggestion
    existing_new_parent = @appeal.reload.tasks.open.find { |open_task| open_task.type == parent.type }
```

Reply:
Alternative:
```suggestion
    existing_new_parent = @appeal.reload.tasks.open.of_type(parent.type).first
```

Also see if you want to instead use https://github.com/department-of-veterans-affairs/caseflow/blob/daf0ce33736db3123659d8fa0f8fa9ae58e5fc63/app/models/task.rb#L669
or https://github.com/department-of-veterans-affairs/caseflow/blob/daf0ce33736db3123659d8fa0f8fa9ae58e5fc63/app/models/task.rb#L445